### PR TITLE
Add order model

### PR DESCRIPTION
**Pull request by waspp01**

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management", description = "Endpoints for managing shopping cart")
+@RestController
+@RequestMapping("/cart")
```
Comment (Rommelua) [inline src/main/java/com/example/onlinebookstore/controller/ShoppingCartController.java]:
```suggestion
@RequestMapping("/carts")
```

File: src/main/java/com/example/onlinebookstore/controller/ShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+@RequiredArgsConstructor
+public class ShoppingCartController {
```
Comment (Rommelua) [inline src/main/java/com/example/onlinebookstore/controller/ShoppingCartController.java]: ShoppingCartController is not a part of this task

Comment (sokimaaa) [reply src/main/java/com/example/onlinebookstore/controller/ShoppingCartController.java]: not fixed

Comment (koval-kn) [reply src/main/java/com/example/onlinebookstore/controller/ShoppingCartController.java]: not fixed

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/onlinebookstore/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.List;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "orders")
+public class Order {
```
Comment (koval-kn) [inline src/main/java/com/example/onlinebookstore/model/Order.java:31]: soft-delete approach was not implemented
image

File: src/main/java/com/example/onlinebookstore/model/OrderItem.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "order_items")
+public class OrderItem {
```
Comment (koval-kn) [inline src/main/java/com/example/onlinebookstore/model/OrderItem.java:24]: soft-delete approach was not implemented
image

File: src/main/java/com/example/onlinebookstore/controller/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        User user = (User) authentication.getPrincipal();
+        return orderService.createOrder(user, requestDto);
+    }
+
+    @GetMapping
+    @Operation(summary = "Get orders history",
+            description = "Get all info about the orders")
+    @ApiResponse(responseCode = "200", description = "Order information",
+            content = {@Content(mediaType = "application/json")})
+    public List<OrderDto> getAll(Authentication authentication) {
```
Comment (koval-kn) [inline src/main/java/com/example/onlinebookstore/controller/OrderController.java]: pagination was not implemented
image

File: src/main/java/com/example/onlinebookstore/service/OrderService.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import com.example.onlinebookstore.dto.order.OrderDto;
+import com.example.onlinebookstore.dto.orderitem.OrderItemDto;
+import com.example.onlinebookstore.model.Order;
+import com.example.onlinebookstore.model.User;
+import java.util.List;
+
+public interface OrderService {
+    OrderDto createOrder(User user, CreateOrderRequestDto requestDto);
+
+    List<OrderDto> getAll(Long userId);
```
Comment (koval-kn) [inline src/main/java/com/example/onlinebookstore/service/OrderService.java]: pagination was not implemented
image

File: src/main/java/com/example/onlinebookstore/model/OrderItem.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    @JoinColumn(name = "book_id", nullable = false)
+    private Book book;
+
+    @Column(nullable = false)
+    private Integer quantity;
+
+    @Column(nullable = false)
+    private BigDecimal price;
+
+
```
Comment (koval-kn) [inline src/main/java/com/example/onlinebookstore/model/OrderItem.java]: redundant blank lines
```suggestion
```

File: src/main/java/com/example/onlinebookstore/service/impl/OrderServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+        order.setTotal(total);
+        order.setShippingAddress(requestDto.getShippingAddress());
+        Order saved = orderRepository.save(order);
+        shoppingCartService.cleanShoppingCart(shoppingCart.getId());
+        return orderMapper.toDto(saved);
+    }
+
+    @Override
+    public List<OrderDto> getAll(Long userId) {
```
Comment (koval-kn) [inline src/main/java/com/example/onlinebookstore/service/impl/OrderServiceImpl.java]: pagination was not implemented
image

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/onlinebookstore/service/OrderItemService.java [hunk 9]
```diff
@@ -0,0 +1,4 @@
+package com.example.onlinebookstore.service;
+
+public interface OrderItemService {
```
Comment (Rommelua) [inline src/main/java/com/example/onlinebookstore/service/OrderItemService.java]: What is it for?

File: src/main/java/com/example/onlinebookstore/service/impl/OrderItemServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,4 @@
+package com.example.onlinebookstore.service.impl;
+
+public class OrderItemServiceImpl {
```
Comment (Rommelua) [inline src/main/java/com/example/onlinebookstore/service/impl/OrderItemServiceImpl.java]: ???

**Pull request merged**